src/Pages/Register/Register.tsx

```diff
@@ -1,10 +1,191 @@
+} from "@mui/material";
+import axios from "axios";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import baseUrl from "../../utils/Custom/Custom";
+import "./Register.module.scss";
 
 const Register = () => {
-  return <>
-    <div>Register</div>
-  </>
-}
+  const navigate = useNavigate();
 
-export default Register
\ No newline at end of file
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const onSubmit = (data) => {
+    data.role = "user";
+    const addFormData = new FormData();
+    addFormData.append("userName", data["userName"]);
+    addFormData.append("email", data["email"]);
+    addFormData.append("password", data["password"]);
+    addFormData.append("confirmPassword", data["confirmPassword"]);
+    addFormData.append("phoneNumber", data["phoneNumber"]);
+    addFormData.append("country", data["country"]);
+    addFormData.append("role", data["role"]);
+    addFormData.append("profileImage", data["profileImage"][0]);
+
+    axios
+      .post(`${baseUrl}/api/v0/portal/users`, addFormData)
+      .then((res) => {
+        console.log("data", data);
+        toast.success("created successfully");
+        navigate("/login");
+      })
+      .catch((err) => {
+        console.log(err.response.data.message);
+        toast.error(err.message);
+      });
+  };
+  return (
+    <>
+      <Box component="div">
+        {" "}
+        <Typography variant="h4" component="h4" sx={{ padding: "20px" }}>
+          <Box component="span" sx={{ color: "skyblue" }}>
+            Stay
+          </Box>
+          cation.
+        </Typography>
+      </Box>
+      <Box sx={{ padding: "30px 70px" }}>
+        <Box component="div">
+          <Typography variant="h4" component="h4">
+            Sign up
+          </Typography>
+          <Typography>
+            If you already have an account register
+            <br /> You can
+            <Link to="/login"> Login here !</Link>
+          </Typography>
+        </Box>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <InputLabel sx={{ marginTop: "5px" }}>User Name</InputLabel>
+          <FormControl sx={{ width: "100%" }}>
+            <OutlinedInput
+              type="text"
+              placeholder="Please type here ..."
+              {...register("userName", {
+                required: true,
+              })}
+            />
+            {errors.userName && errors.userName.type === "required" && (
+              <Box className="text-danger" sx={{ color: "red" }}>
+                userName is required
+              </Box>
+            )}
+          </FormControl>
+          <Grid container spacing={2}>
+            <Grid item md={6} sm={12}>
+              <InputLabel sx={{ marginTop: "5px" }}>Phone Number</InputLabel>
+              <FormControl sx={{ width: "100%" }}>
+                <OutlinedInput
+                  type="number"
```

Comment:
Phone number should be a string, not a number, since some formats include the +,-, and spaces and some start with 0 which will be ignored as a number.